lib/sshutils/sftp/sftp_test.go

```diff
@@ -585,7 +584,7 @@ func TestHomeDirExpansion(t *testing.T) {
 			name: "~user path",
 			path: "~user/foo",
 			errCheck: func(t require.TestingT, err error, i ...interface{}) {
-				require.True(t, trace.IsBadParameter(err))
+				require.ErrorAs(t, err, &PathExpansionError{path: "~user/foo"})
```

Comment:
nit: using ErrorAs with a specific path is kind of pretending that the test asserts that the path matches, when it actually doesn't. Because PathExpansionErrors are values and not pointers, you can actually use ErrorIs here
```suggestion
				require.ErrorIs(t, err, PathExpansionError{path: "~user/foo"})
```